docs/pages/dev/how_to/use_plugins/use_plugins.en.md

```diff
@@ -0,0 +1,369 @@
+- `homeOffline`: Component to display when the plugin is offline
+- `key`: Unique identifier for the plugin
+- `redirections`: Array of routing redirections
+- `theme`: Custom theme for the plugin
+- `customComponents`: Array of custom components to be registered
+
+#### Example Configuration
+
+```typescript
+// js/config.tsx
+import React from 'react';
+import MyPluginComponent from './components/MyPluginComponent';
+import MyPluginHome from './components/MyPluginHome';
+
+export default {
+    routes: [
+        {
+            path: '/my-plugin',
+            component: MyPluginComponent,
+            permissions: ['plugin_access'],
+        },
+    ],
+    menu: [
+        {
+            label: 'My Plugin',
+            icon: 'plugin',
+            path: '/my-plugin',
+            permissions: ['plugin_access'],
+            subMenu: [
+                {
+                    label: 'Sub Feature',
+                    path: '/my-plugin/sub-feature',
+                    permissions: ['sub_feature_access'],
+                },
+            ],
+        },
+    ],
+    translations: {
+        en: {
+            'myPlugin.title': 'My Plugin',
+            'myPlugin.description': 'This is my plugin',
+        },
+        fr: {
+            'myPlugin.title': 'Mon Plugin',
+            'myPlugin.description': 'C\'est mon plugin',
+        },
+    },
+    homeUrl: '/my-plugin',
+    homeOnline: MyPluginHome,
+    key: 'my-plugin',
+    baseUrls: {
+        api: '/api/my-plugin',
+    },
+    paramsConfig: {
+        'my-plugin': ['id', 'type'],
+    },
+    redirections: [
+        {
+            from: '/old-path',
+            to: '/my-plugin',
+        },
+    ],
+    customComponents: [
+        {
+            key: 'myCustomComponent',
+            component: MyCustomComponent,
+        },
+    ],
+};
+```
+
+## Development vs Production
+
+### Development Mode
+
+In development (`webpack.dev.js`):
+- Files are generated on-the-fly when changes are detected
+- Hot Module Replacement (HMR) is enabled for quick development
+- Source maps are available for debugging
+- Plugin changes are automatically detected and reloaded
+
+### Production Mode
+
+In production (`webpack.prod.js`):
+- Files are generated once during the build process
+- Code is minified and optimized
+- Dead code elimination is performed
+- Source maps are optional (configurable)
+
+## Loading Plugins at Runtime
+
+The main application:
+1. Loads the plugin keys from `IasoModules/plugins/keys`
+2. Loads configurations from `IasoModules/plugins/configs`
+3. Registers plugin routes in the router
+4. Adds plugin menu items to the navigation
+5. Initializes plugin components and themes
+
+## Creating a New Plugin
+
+Quick start guide:
+
+1. Create a new directory in the plugins folder:
+   ```bash
+   mkdir -p plugins/my-plugin/js
+   ```
+
+2. Create the basic file structure:
+   ```bash
+   cd plugins/my-plugin
+   touch js/config.tsx js/index.tsx README.md
+   ```
```

Comment:
We should make a script for that, that would also create the translations files and folders. We could then also impose some additional structure like the `domains` folder. (new ticket?)

Reply:
yes, the Pr is already huge